libs/application/templates/inheritance-report/src/fields/HeirsAndPartitionRepeater/index.tsx

```diff
@@ -186,12 +152,81 @@ export const HeirsAndPartitionRepeater: FC<
     setTotal(total)
   }, [getValues, id, props.sumField, setValue])
 
-  useEffect(() => {
+  const updateValues = useCallback(
+    (updateIndex: string, value: number) => {
+      const numValue = isNaN(value) ? 0 : value
+      const percentage = numValue > 0 ? numValue / 100 : 0
+
+      const currentHeir = getValueViaPath(answers, updateIndex) as EstateMember
+
+      const currentNationalId = valueToNumber(currentHeir?.nationalId)
+
+      const heirs = getValues()?.heirs?.data as EstateMember[]
+
+      // currently we can only check if heir is spouse by relation string value...
+      const spouse = (heirs ?? []).filter(
+        (heir) => heir.relation === 'Maki' || heir.relation === 'Spouse',
+      )
+
+      let isSpouse = false
+
+      // it is not possible to select more than one spouse but for now we will check for it anyway
+      if (spouse.length > 0) {
+        spouse.forEach((currentSpouse) => {
+          isSpouse =
+            valueToNumber(currentSpouse?.nationalId) === currentNationalId
+        })
+      }
+
+      const netPropertyForExchange = Number(
+        getValueViaPath(answers, 'netPropertyForExchange'),
+      )
+      const inheritanceValue = Math.round(netPropertyForExchange * percentage)
+
+      const taxFreeInheritanceValue = isSpouse
+        ? inheritanceValue
+        : Math.round(TAX_FREE_LIMIT * percentage)
```

Comment:
Isn't the TAX_FREE_LIMIT in the externalData now? Look for `inheritanceTax` or whatever. It should be there somehwere :sweat_smile: